### PR TITLE
Add configuration file

### PR DESCRIPTION
**Pull request by janedbal**
- Resolves https://github.com/shipmonk-rnd/name-collision-detector/issues/20

### CODE REVIEW

File: src/DetectionConfig.php [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+            if ($absoluteRealPath === false) {
+                throw new LogicException("Unable to realpath \"$path\" even though it is existing file or dir");
+            }
+
+            return $absoluteRealPath;
+        };
+
+        $this->scanPaths = array_map($normalizePath, $scanPaths);
+        $this->excludePaths = array_map($normalizePath, $excludePaths);
+        $this->currentDirectory = $currentDirectory;
```
Comment (olsavmic) [inline src/DetectionConfig.php:85]: Current DIR is not normalized so I suppose it should be validated that it's an absolute path.

File: src/DetectionConfig.php [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+        $configData = file_get_contents($configFilePath);
+
+        if ($configData === false) {
+            throw new InvalidConfigException("Failure while opening config file $configFilePath");
+        }
+
+        try {
+            /** @throws JsonException */
+            $configArray = json_decode($configData, true, JSON_PRESERVE_ZERO_FRACTION | 4194304); // throw on error if available
```
Comment (olsavmic) [inline src/DetectionConfig.php:116]: Extract the constant to local JSON_THROW_ON_ERROR with explanation.

File: README.md [hunk 3]
```diff
@@ -27,6 +27,18 @@ GlobalInterface1 is defined 2 times:
  > /tests/sample-collisions/file2.php
 ```
 
+## Configuration:
+If file named `collision-detector.json` is present within current working directory, its contents are taken as configuration options. Possible config options:
+```json5
+{
+    "scanPaths": ["src", "tests"], // files/directories to scan, relative to config file directory
```
[the remark this answers is not in the document]
Comment (janedbal) [reply README.md:34]: No, I want to keep it simple unless somebody comes with a usecase hard to solve with that. I even believe most users dont need to use config file at all.

[the remark this answers is not in the document]
Comment (olsavmic) [reply README.md:34]: Then please document it here in readme

**Review by olsavmic — APPROVED** [review]

**Pull request merged**